src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -43,73 +45,88 @@
- * Additionally, access to a memory segment is subject to the thread-confinement checks enforced by the owning memory
- * session; that is, if the segment is associated with a shared session, it can be accessed by multiple threads;
- * if it is associated with a confined session, it can only be accessed by the thread which owns the memory session.
+ * Every memory segment has an {@linkplain #address() address}, expressed as a {@code long} value.
+ * The nature of a segment's address depends on the kind of the segment:
+ * <ul>
+ * <li>The address of a heap segment is not a physical address, but rather an offset within the region of memory
+ * which backs the segment. The region is inside the Java heap, so garbage collection might cause the region to be
+ * relocated in physical memory over time, but this is not exposed to clients of the {@code MemorySegment} API who
+ * see a stable <em>virtualized</em> address for a heap segment backed by the region.
+ * A heap segment obtained from one of the {@link #ofArray(int[])} factory methods has an address of zero.</li>
+ * <li>The address of a native segment (including mapped segments) denotes the physical address of the region of
+ * memory which backs the segment.</li>
+ * </ul>
+ * <p>
+ * Every memory segment has a {@linkplain #byteSize() size}. The size of a heap segment is derived from the Java array
+ * from which it is obtained. This size is predictable across Java runtimes.
+ * The size of a native segment is either passed explicitly
+ * (as in {@link MemorySegment#allocateNative(long, MemorySession)}) or derived from a {@link MemoryLayout}
+ * (as in {@link MemorySegment#allocateNative(MemoryLayout, MemorySession)}). The size of a memory segment is typically
+ * a positive number but may be <a href="#wrapping-addresses">zero</a>, but never negative.
  * <p>
- * Heap segments are always associated with the {@linkplain MemorySession#global() global} memory session.
- * This session cannot be closed, and segments associated with it can be considered as <em>always alive</em>.
- * Buffer segments are typically associated with the global memory session, with one exception: buffer segments created
- * from byte buffer instances obtained calling the {@link #asByteBuffer()} method on a memory segment {@code S}
- * are associated with the same memory session as {@code S}.
+ * The address and size of a memory segment jointly ensure that access operations on the segment cannot fall
+ * <em>outside</em> the boundaries of the region of memory which backs the segment.
+ * That is, a memory segment has <em>spatial bounds</em>.
+ * <p>
+ * Every memory segment is associated with a {@linkplain MemorySession memory session}. This ensures that access operations
+ * on a memory segment cannot occur when the region of memory which backs the memory segment is no longer available
+ * (e.g. after the memory session associated with the accessed memory segment is no longer {@linkplain MemorySession#isAlive() alive}.
```

Comment:
Missing close brace:
```suggestion
 * (e.g. after the memory session associated with the accessed memory segment is no longer {@linkplain MemorySession#isAlive() alive}).
```